### ISSUE TITLE
Action Replays defaults Image depth to 12-bit on application startup #34

### ISSUE THREAD

**Issue by OfficerArtoria** [0xC0000054/avif-format#11]
I have saved an action that ends with a file save to AVIF with Image depth set to 8bit. The action itself works well in most circumstances, except whenever I start my Photoshop up, it defaults to 12-bit. This requires re-recording the action again, which would be default back to 8-bit. Other parameters are working for me properly for Action Replay.

**Comment by 0xC0000054** [comment]
The plugin defaults to 12-bit when Photoshop is first started. The last used bit-depth setting from Actions should override that default.
I am not sure what the issue is, and I currently do not have access to a machine with Photoshop to test it.

Is your document 8-bit?

The plugin can override the image mode for certain 32-bit HDR save configurations, but that code should be disabled for 8-bit and 16-bit documents.

**Comment by OfficerArtoria** [comment]
In the action recording, I've explicitly selected to save the file as 8-bit. Subsequent uses of this action on the same instance works as intended.

The issue is, when I close Photoshop, and open it again, the action itself doesn't save as 8-bit (and I would assume it goes back to defaulting to 12-bit).

The documents I am using are all in 8-bit, I can confirm that.

**Comment by 0xC0000054** [comment]
The bit depth is saved in the Action data. It should be visible in the Actions palette as the 'image depth' item.
I do not know why it would be exhibiting that behavior when restarting Photoshop,

Could you upload a zip containing one of the Actions that is showing this issue?

I should be able to see the image mode that was saved when viewing the file in a hex editor.

**Comment by OfficerArtoria** [comment]
Here is the file containing the action. I do suspect that the bit depth is being saved correctly as 8-bit, as when I load photoshop the details of the save action do say explicitly which bit depth it is saving.
https://i.imgur.com/obhmFT4.png
[Save Files.zip](https://github.com/0xC0000054/avif-format/files/11437531/Save.Files.zip)

**Comment by 0xC0000054** [comment]
> I do suspect that the bit depth is being saved correctly as 8-bit, as when I load photoshop the details of the save action do say explicitly which bit depth it is saving.

Looking in a hex editor, the Action file is configured to save 8-bit.

**Comment by OfficerArtoria** [comment]
Is there anything I can do to help you locate a potential issue?

**Comment by 0xC0000054** [comment]
I don't think so. I will need to investigate the issue when debugging the plugin.
